src/components/Modals/ExportModal.tsx

```diff
@@ -70,26 +71,83 @@ export const ExportModal = ({ open, onClose, address }: { open: boolean; onClose
     setDownloadClicked(false);
   }, [data]);
 
-  useEffect(() => {
-    document.addEventListener('keydown', hideOnEscape, true);
-    document.addEventListener('click', hideOnClickOutside, true);
-  }, []);
+  const handleDownload = async () => {
+    if (startDate && endDate) {
+      if (dataSet.value === 'rewards') {
+        setHeaders(['date', 'amount']);
+        const data = await getData(() =>
+          api
+            .get<any>(REACT_APP_DAG_EXPLORER_API_URL + '/' + network + '/validator-nodes/' + address + '/rewards', {
+              startDate: startDate.toLocaleDateString(),
+              endDate: endDate.toLocaleDateString(),
+            })
+            .then((r) => {
+              const data = r.data as {
+                rewards: { rewardAmount: string; date: string }[];
+                isValidator: boolean;
+              };
```

Comment:
```suggestion
            .get<{
              data: {
                rewards: { rewardAmount: string; date: string }[];
                isValidator: boolean;
              };
            }>(REACT_APP_DAG_EXPLORER_API_URL + '/' + network + '/validator-nodes/' + address + '/rewards', {
              startDate: startDate.toLocaleDateString(),
              endDate: endDate.toLocaleDateString(),
            })
            .then((r) => {
              const data = r.data;
```
Be careful with types, especially if you're the one building features that use them ([api.ts](https://github.com/StardustCollective/dag-explorer-v2/blob/6881d7d03250a20401c3c9222dd0faa6e4ee40bd/src/utils/api.ts#L1-L15))